File: Codigo Java/src/Orden.java

package gestorsupermercado;

import java.util.ArrayList;
import java.util.List;

public class Orden {
    private Cliente cliente;
    private List<Producto> items;

    public Orden(Cliente cliente) {
        this.cliente = cliente;
        this.items = new ArrayList<>();
    }

    public void agregarItem(Producto producto) {
        items.add(producto);
    }

    public Cliente getCliente() {
        return cliente;
    }

    public List<Producto> getItems() {
        return items;
    }
}

File: Codigo/orden.py

class ItemOrden:
    def __init__(self, producto, cantidad):
        self.producto = producto
        self.cantidad = cantidad
    
    def calcular_subtotal(self):
        return self.producto.precio * self.cantidad

class Orden:
    def __init__(self, cliente):
        self.cliente = cliente
        self.items = []
        self.total = 0
    
    def agregar_item(self, item):
        self.items.append(item)
        self.total += item.calcular_subtotal()
    
    def calcular_total(self):
        return self.total
